### PR TITLE
feat(psh): Improve psh-init with better logic for custom overrides.

### LINKED ISSUE #134: Clean Up  init-psh & remove build:lib from fresh install
## Description

Describe the issue that you're seeing.

### Steps to reproduce

* Run on a site https://johnsonandjohnson.github.io/Bodiless-JS/#/Development/OtherResources/Platform.sh?id=step-2-initialize-your-project-configuration-files
* Deploy to platform.sh

### Expected result

Deploy of static & edit envs to platform.sh should succeed.
A fresh deploy on edit env should not run build:lib
### Actual result

Issue 1:
* Deploy of static fails on bodiless packages are not found
** https://github.com/johnsonandjohnson/Bodiless-JS/blob/master/packages/bodiless-psh/resources/static/static.platform.app.yaml#L32 should reference canvasx
 
Issue 2:
* Running fresh is killed on edit.env as there are not enough resources
**https://support.platform.sh/hc/en-us/requests/82766

Issue 3:
* logic about the private npm registry should go in a custom.sh & not be in .platform.app.yaml

### Environment

Platform.sh envs - static & edit

### Affected Version

- Bodiless 0.0.39

### PR DESCRIPTION
**Pull request by wodenx**
## Changes
- Provide better mechanism for implementing custom p.sh hooks
- Add logic for setting private npm registry
- move .npmrc creation to build hook
- symlink .cache directory to writable volume in edit container

## Related Issues
Fixes #134 

BREAKING CHANGE: Existing platform.custom.sh implementations will no longer work.

### CODE REVIEW

File: examples/test-site/edit/platform.sh [hunk 1]
```diff
@@ -187,46 +219,27 @@ predeploy () {
...
   pm2 restart backend && pm2 restart frontend || pm2 start ${PLATFORM_APP_DIR}/ecosystem.config.js
 }
 
-if [ "$1" = "deploy" ]; then
-  echo "DEPLOY AT $(date)"
-  predeploy
-  if [ -d ${ROOT_DIR}/.git ]; then
-    # TODO: Install only when necessary.
-    # last_hash=$(git rev-parse HEAD)
-    incremental_deploy
```
Comment (beliayeu) [inline examples/test-site/edit/platform.sh]: @wodenx, am I right that we want to get rid of incremental deployments?

File: examples/test-site/edit/platform.custom.sh [hunk 2]
```diff
@@ -12,24 +12,9 @@
...
+finalize_deploy () {
   npm run bootstrap -- --ci
-elif [ "$1" = "build" ]; then
   npm run build:packages -- --concurrency 1
   npm run build:env-vars
   npm run build:doc
-elif [ "$1" = "finish-deploy" ]; then
-  SITE_DIR=${ROOT_DIR}/examples/test-site
-  cp ${PLATFORM_APP_DIR}/${DEFAULT_ENV} ${SITE_DIR}/.env.development
-  if [ -n "${BV_SCRIPT}" ]; then
```
[the remark this answers is not in the document]
Comment (wodenx) [reply examples/test-site/edit/platform.custom.sh]: @beliayeu afaict the finish-deploy custom hook was never invoked: https://github.com/johnsonandjohnson/Bodiless-JS/tree/master/packages/bodiless-psh/resources/edit -- and i don't think we have BV enabled in the monorepo.   cc @hvanyo

File: examples/test-site/edit/platform.sh (hunk 1 above)
Comment (wodenx) [reply examples/test-site/edit/platform.sh]: There is a story to bring them back, so i left this in as a starting point, but you're right it's currently unused.

File: packages/bodiless-psh/resources/static/static.platform.app.yaml [hunk 3]
```diff
@@ -23,56 +23,11 @@ type: 'nodejs:10'
 build:
     flavor: none
 
-# The hooks that will be triggered when the package is deployed.
 hooks:
```
[the remark this answers is not in the document]
Comment (wodenx) [reply packages/bodiless-psh/resources/static/static.platform.app.yaml]: root .platform.app.yaml is really a one-off and doen't participate in our inheritance scheme (is not copied during `npm psh-update`.  Also eventually I'd like to deprecate it, and deploy the test-site like any other site (using local packages and travis, probably).  So I decided to leave as is for now.

File: examples/test-site/edit/platform.custom.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (hvanyo) [reply examples/test-site/edit/platform.custom.sh]: Correct BV is not enabled in monorepo or it would expose J&J configuration.

**Pull request merged**